samcli/lib/cookiecutter/interactive_flow_creator.py

```diff
@@ -0,0 +1,125 @@
+import yaml
+from samcli.commands.exceptions import UserException
+from samcli.yamlhelper import parse_yaml_file
+from .interactive_flow import InteractiveFlow
+from .question import Question, QuestionKind
+
+
+class QuestionsNotFoundException(UserException):
+    pass
+
+
+class QuestionsFailedParsingException(UserException):
+    pass
+
+
+class InteractiveFlowCreator:
+    @staticmethod
+    def create_flow(flow_definition_path: str, extra_context: Optional[Dict] = None):
+        """
+        This method parses the given json/yaml file to create an InteractiveFLow. It expects the file to define
+        a list of questions. It parses the questions and add it to the flow in the same order they are defined
+        in the file, i.e. the default next-question of a given question will be the next one defined in the file,
+        while also respecting the question-defined _next_question map if provided.
+
+        Parameters:
+        ----------
+        flow_definition_path: str
+            A path to a json/yaml file that defines the questions of the flow. the file is expected to be in the
+            following format:
+            {
+                "Questions": [
+                    {
+                      "key": "key of the corresponding cookiecutter config",
+                      "question": "the question to prompt to the user",
+                      "kind": "the kind of the question, for example: confirm",
+                      "isRequired": true/false,
+                      # optional branching logic to jump to a particular question based on the answer. if not given
+                      # will automatically go to next question
+                      "nextQuestion": {
+                        "True": "key of the question to jump to if the user answered 'Yes'",
+                        "False": "key of the question to jump to if the user answered 'Yes'",
+                      }
+                    },
+                    ...
+                ]
+            }
+        extra_context: Dict
+            if the template contains variable($variableName) this parameter provides the values for those variables.
+
+        Returns: InteractiveFlow(questions={k1: q1, k2: q2, ...}, first_question_key="first question's key")
+        """
+        questions, first_question_key = InteractiveFlowCreator._load_questions(flow_definition_path, extra_context)
+        return InteractiveFlow(questions=questions, first_question_key=first_question_key)
+
+    @staticmethod
+    def _load_questions(
+        flow_definition_path: str, extra_context: Optional[Dict] = None
+    ) -> Tuple[Dict[str, Question], str]:
+        previous_question: Optional[Question] = None
+        first_question_key: str = ""
+        questions: Dict[str, Question] = {}
+        questions_definition = InteractiveFlowCreator._parse_questions_definition(flow_definition_path, extra_context)
+
+        for question in questions_definition.get("Questions"):
```

Comment:
UX: `Questions` isn't consistent with other capitalization. Better to keep it as `questions` so that we don't have to remember which key is the exception.